### PR TITLE
Broadcasting a private FM radio station using a Raspberry pi

### PR DESCRIPTION
**Pull request by ruphus-muita**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by ahmadmardeni1** [comment]
Good afternoon and thank you for submitting your article for publishing.
Your article has been entered into our queue and should be reviewed (and published pending any changes) as soon as a peer reviewer is finished with the ones in the queue before it. :rocket:
Thank you. @ruphus-muita

**Comment by ruphus-muita** [comment]
@ahmadmardeni1 Well noted

File: content/articles/broadcasting-a-pirate-fm-radio-station-using-a-raspberry-pi/index.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+description: This article will explain how one can create a private FM radio with the use of a Raspberry Pi and internet connection. The private FM radio broadcast can reach a radius of around fifty meters.
+author: ruphus-muita
+date: 2021-07-19T00:00:00-18:00
+topics: [Networking]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/broadcasting-a-pirate-fm-radio-station-using-a-raspberry-pi/hero.jpg
+    alt: Raspberry Pi Image 
+---
```
Comment (Bayler) [inline content/articles/broadcasting-a-pirate-fm-radio-station-using-a-raspberry-pi/index.md:16]: Hi @ruphus-muita this was unnecessary. Please don't add it next time. It's called 'front matter' and it's the job of a peer reviewer to add them. As peer reviewers, we also add a 'Peer Reviewers Contributors (PRC) line' at the bottom of the article. You'll see it once I'm done reviewing your article. Don't add the two. You'll get the hang of it with time. Don't worry.

**Comment by Bayler** [comment]
Hi @ruphus-muita Welcome to Section's Engineering Program. You've just successfully submitted your first PR. Congratulations! We're also glad to have you. 

Before I can continue reviewing, please remove the front matter as commented earlier. Also, make sure in your author bio information, the "Avatar.jpg" is renamed to "avatar.jpg". It shouldn't be in caps.

**Comment by hectorkambow** [comment]
plagiarism check done

File: content/articles/broadcasting-a-pirate-fm-radio-station-using-a-raspberry-pi/index.md (hunk 1 above)
Comment (ruphus-muita) [reply content/articles/broadcasting-a-pirate-fm-radio-station-using-a-raspberry-pi/index.md]: @Bayler Thanks for the insight. This is done.

**Comment by ruphus-muita** [comment]
> Hi @ruphus-muita Welcome to Section's Engineering Program. You've just successfully submitted your first PR. Congratulations! We're also glad to have you.
> 
> Before I can continue reviewing, please remove the front matter as commented earlier. Also, make sure in your author bio information, the "Avatar.jpg" is renamed to "avatar.jpg". It shouldn't be in caps.

@Bayler This is done.

**Comment by Bayler** [comment]
> Hi @ruphus-muita Welcome to Section's Engineering Program. You've just successfully submitted your first PR. Congratulations! We're also glad to have you.
> 
> Before I can continue reviewing, please remove the front matter as commented earlier. Also, make sure in your author bio information, the "Avatar.jpg" is renamed to "avatar.jpg". It shouldn't be in caps.

@ruphus-muita how about the Avatar.jpg?

File: content/articles/broadcasting-a-pirate-fm-radio-station-using-a-raspberry-pi/index.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
+Setting up a private radio station is easy using a Raspberry Pi. This process should take thirty to sixty minutes to complete. By the end of this hour, you should be able to broadcast within a radius of around fifty meters. However, this is also facilitated by the use of a good antenna. 
+<div style="text-align: center; padding-bottom: 5%;">
+<img style="padding: 0; margin:0;" src="/engineering-education/broadcasting-a-pirate-fm-radio-station-using-a-raspberry-pi/raspberry21.jpg" alt="Raspberry Pi diagram"><br>
+<span style="font-size: 11px;">Figure 1: Sample Rapberry Pi <a href="https://unsplash.com/photos/raLeFIxXgDY">Raspberry Pi</a>, 2021.<span>
+</div>
```
Comment (Bayler) [inline content/articles/broadcasting-a-pirate-fm-radio-station-using-a-raspberry-pi/index.md:5]: Here, I'd like you to be creative and put a picture of a raspberry pi with connections coming out of the GPIO pins and the antenna connected to it. You don't have to get it through Unsplash. You could build one yourself (which i assume you have) and take a photo of it. That would really grab the readers attention.  
Secondly, the format you've used there is wrong. Please use the format below:

For example, yours should be:

Raspberry Pi diagram

*[Image Source: Unsplash](https://unsplash.com/photos/raLeFIxXgDY)*

File: content/articles/broadcasting-a-pirate-fm-radio-station-using-a-raspberry-pi/index.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+Materials required for this tutorial include:
+1.	Raspberry Pi.
```
Comment (Bayler) [inline content/articles/broadcasting-a-pirate-fm-radio-station-using-a-raspberry-pi/index.md:8]: You have to be more specific on which Raspberry Pi we are using. i.e., Raspberry pi 3, 4?

File: content/articles/broadcasting-a-pirate-fm-radio-station-using-a-raspberry-pi/index.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+2.	Microphone.
+3.	Internet connection.
+
+### Prerequisites
+In this guide, it is assumed that the reader has a Raspberry Pi that has a running operating system. We will assume that the running Raspberry Pi. It should also have an active connection to the internet. Additionally, you must have access to your Pi, through whichever means you prefer. It could be through a terminal or by a virtual server such as VNC. This guide will use a Putty program to run commands on our Pi. This guide is also based on **Rasbian Jessie installed Raspberry Pi 3**. 
+
```
Comment (Bayler) [inline content/articles/broadcasting-a-pirate-fm-radio-station-using-a-raspberry-pi/index.md:14]: Please add the link to "Rasbian Jessie installed Raspberry Pi 3"

File: content/articles/broadcasting-a-pirate-fm-radio-station-using-a-raspberry-pi/index.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+This article will cover:
+- [How the private FM broadcast will work](#how-the-private-fm-broadcast-will-work)
+- [Setting up the Raspberry](#setting-up-the-raspberry)
+- [Conversion of PRi to Private FM transmitter](#conversion-of-pri-to-private-fm-transmitter)
+- [Conclusion](#conclusion)
+- [Relevant resources](#relevant-resources)  
+  
+### How the private FM broadcast will work
+To reduce electromagnetic interference, all microprocessors are fitted with a synchronous digital system. Using a spread-spectrum clock signal or what is referred to as SSCS in short form, suppression of electromagnetic interference is achieved. This signal has a frequency of between 1MHz and 250MHz which falls between the FM band. Therefore, by using a program that modulates frequency, the Pi will be tweaked to operate as a radio transmitter. This signal will then be produced via the GPIO pin four of the Pi. A short wire of around twenty centimeters can be attached to act as an antenna.
+
```
Comment (Bayler) [inline content/articles/broadcasting-a-pirate-fm-radio-station-using-a-raspberry-pi/index.md:25]: It'll be good to educate the reader on what frequency constitutes the FM band. Is it 88Mhz to 108Mhz? Include that in your article

File: content/articles/broadcasting-a-pirate-fm-radio-station-using-a-raspberry-pi/index.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+### Setting up the Raspberry
+Since the Raspberry Pi has a running operating system, boot it and connect the output HDMI to a monitor. Also connect a keyboard and mouse. On accessing the Desktop, search for the network option which will enable connection to the router (internet). Next, move on to the pi configuration through the menu and enable SSH communication. 
+Once this is enabled, we can now access the Raspberry using Putty as long the personal computer and the Pi are connected to the same network. Open Putty on the personal computer and key in the IP address of the Raspberry Pi to connect via SSH. A session will then appear and ask for a login credentials. The credentials to be used here are pi for username and raspberry for password.
```
Comment (Bayler) [inline content/articles/broadcasting-a-pirate-fm-radio-station-using-a-raspberry-pi/index.md:28]: This is the part where I need you to include screenshots of what you're explaining

File: content/articles/broadcasting-a-pirate-fm-radio-station-using-a-raspberry-pi/index.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+### Setting up the Raspberry
+Since the Raspberry Pi has a running operating system, boot it and connect the output HDMI to a monitor. Also connect a keyboard and mouse. On accessing the Desktop, search for the network option which will enable connection to the router (internet). Next, move on to the pi configuration through the menu and enable SSH communication. 
+Once this is enabled, we can now access the Raspberry using Putty as long the personal computer and the Pi are connected to the same network. Open Putty on the personal computer and key in the IP address of the Raspberry Pi to connect via SSH. A session will then appear and ask for a login credentials. The credentials to be used here are pi for username and raspberry for password. 
+
+### Conversion of PRi to Private FM transmitter
+*Step One:*
```
Comment (Bayler) [inline content/articles/broadcasting-a-pirate-fm-radio-station-using-a-raspberry-pi/index.md:31]: For a second I was wondering what's PRi? Please explain what PRi is in one sentence before getting started with Step one

File: content/articles/broadcasting-a-pirate-fm-radio-station-using-a-raspberry-pi/index.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+```
+*Step Two:*
+
+This stage involves cloning the code from GitHub. In order to achieve this, we use the git clone commands as shown below.
+```JSON
+sudo git clone https://github.com/markondej/fm_transmitter
+```
+*Step Three:*
+
+This code is written in C. This, therefore, calls for the use of compilers to compile it. In this guide, we will use a program known as gcc and g++. The compiler to be used is known as *make*. The following commands will be used to download the compilers.
```
Comment (Bayler) [inline content/articles/broadcasting-a-pirate-fm-radio-station-using-a-raspberry-pi/index.md:46]: In this section, we are compiling the code written in C to which language? Please make it clear for the reader

**Comment by Bayler** [comment]
This was a interesting read @ruphus-muita Please work on the requested revisions before we can proceed.

**Comment by ruphus-muita** [comment]
> This was an interesting read @ruphus-muita Please work on the requested revisions before we can proceed.

@Bayler Thanks. Working on them

**Comment by ruphus-muita** [comment]
> This was an interesting read @ruphus-muita Please work on the requested revisions before we can proceed.

Hi, @Bayler I have uploaded the changes. Kindly verify

**Comment by Bayler** [comment]
> > This was an interesting read @ruphus-muita Please work on the requested revisions before we can proceed.
> 
> Hi, @Bayler I have uploaded the changes. Kindly verify

Where have you uploaded the changes? The article is still the way it was

**Comment by Bayler** [comment]
@ruphus-muita Are you having trouble uploading your revisions? You've uploaded unnecessary files. Please delete them

**Comment by ruphus-muita** [comment]
> @ruphus-muita Are you having trouble uploading your revisions? You've uploaded unnecessary files. Please delete them

@Bayler Sorry. I am having trouble reverting the process.

**Comment by ruphus-muita** [comment]
> > @ruphus-muita Are you having trouble uploading your revisions? You've uploaded unnecessary files. Please delete them
> 
> @Bayler Sorry. I am having trouble reverting the process.
> 

@Bayler Kindly, I need your assistance. How can I delete the extra files?

**Comment by Bayler** [comment]
> > @ruphus-muita Are you having trouble uploading your revisions? You've uploaded unnecessary files. Please delete them
> 
> @Bayler Sorry. I am having trouble reverting the process.

@ahmadmardeni1 @hectorkambow @LinusMuema @louisefindlay23 any suggestion on why this might be happening and how to revert it without messing up the article?

**Comment by Bayler** [comment]
> > > @ruphus-muita Are you having trouble uploading your revisions? You've uploaded unnecessary files. Please delete them
> > 
> > 
> > @Bayler Sorry. I am having trouble reverting the process.
> 
> @Bayler Kindly, I need your assistance. How can I delete the extra files?

Please allow me to seek assistance.

**Comment by ahmadmardeni1** [comment]
I'm not sure why this PR contains hundreds of commits. @Bayler 

Can you please open a new PR? Make sure you use a separate branch on your forked repo for article submission, not the master branch.  @ruphus-muita

**Comment by ruphus-muita** [comment]
Okay. Thanks

**Comment by Bayler** [comment]
> Okay. Thanks @ahmadmardeni1

Also make sure that the new article to be uploaded in the new PR is the updated version of the article with the requested changes

**Comment by ruphus-muita** [comment]
Okay. Working on it

**Comment by ahmadmardeni1** [comment]
@ me and Bayler when you open the new PR - I'm closing this one. @ruphus-muita

**Pull request closed without merging**